### PR TITLE
Update admin.py

### PR DESCRIPTION
**Pull request by ozeranskiy**
add extra_context to history_form_view

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by rossmechanic** [comment]
@ozeranskiy can you add some more context about this change? Also, add the change to `CHANGES.rst` and add your name to `AUTHORS.rst`

**Comment by ozeranskiy** [comment]
> @ozeranskiy can you add some more context about this change? Also, add the change to `CHANGES.rst` and add your name to `AUTHORS.rst`

Done.
Usage example:
```python
class SimpleHistoryAdminMixin(SimpleHistoryAdmin):
    def history_form_view(self, request, object_id, version_id, extra_kwargs=None):
        extra_kwargs = extra_kwargs or {}
        extra_kwargs['obj'] = 'My data'
        return super(SimpleHistoryAdminMixin, self).history_form_view(
            request, object_id, version_id, extra_context=extra_kwargs
        )
```

File: simple_history/tests/tests/test_admin.py [hunk 1]
```diff
@@ -467,7 +468,9 @@ def test_history_form_view_without_getting_history(self):
             'save_on_top': admin.save_on_top,
             'root_path': getattr(admin_site, 'root_path', None),
         }
+        extra_context = {}
```
Comment (rossmechanic) [inline simple_history/tests/tests/test_admin.py]: Why do you need to set `extra_context = {}` here?

Comment (ozeranskiy) [reply simple_history/tests/tests/test_admin.py]: in the test it is not necessary.

Comment (rossmechanic) [reply simple_history/tests/tests/test_admin.py]: Shouldn't the test pass with `extra_context=None` as the default? Or do you want to set the default to `extra_context={}`?

Comment (ozeranskiy) [reply simple_history/tests/tests/test_admin.py]: In the test, I want to set the default extra_context={}
Your option will work too. extra_context in the test does not affect the functionality

Comment (rossmechanic) [reply simple_history/tests/tests/test_admin.py]: Won't the test still pass if you don't have this line? Essentially this makes line 473 irrelevant.

Comment (ozeranskiy) [reply simple_history/tests/tests/test_admin.py]: I add extra_context in context. In this case, I do not pass the custom parameters and extra_context = {} can be omitted. Should I delete this line in tests?

Comment (rossmechanic) [reply simple_history/tests/tests/test_admin.py]: Yes delete this line and then I'll approve and merge

Comment (ozeranskiy) [reply simple_history/tests/tests/test_admin.py]: Done :)

File: simple_history/admin.py [hunk 2]
```diff
@@ -107,7 +107,8 @@ def response_change(self, request, obj):
             return super(SimpleHistoryAdmin, self).response_change(
                 request, obj)
 
-    def history_form_view(self, request, object_id, version_id):
+    def history_form_view(self, request, object_id, version_id,
+                          extra_context=None):
```
Comment (rossmechanic) [inline simple_history/admin.py]: `extra_context` is `None` by default, so you shouldn't have to add it here. Try to add separate tests where `extra_context` has an actual value and passes through

**Comment by rossmechanic** [comment]
Hey @ozeranskiy are you still working on this or want me to take it over?

**Comment by ozeranskiy** [comment]
> Hey @ozeranskiy are you still working on this or want me to take it over?


In the near future I can not write a test. Can you accept the current state?

**Comment by barm** [comment]
I can take it over and add some additional tests this week. I'm not a fan of merge now and add tests later because it's easy to lose track of that work. Thanks for your contribution!

**Comment by ozeranskiy** [comment]
> I can take it over and add some additional tests this week. I'm not a fan of merge now and add tests later because it's easy to lose track of that work. Thanks for your contribution!

ok, thanks you.

**Comment by barm** [comment]
Opened #499 so I'll close this. Thanks again!

**Pull request closed without merging**